src/ModValue.cpp

```diff
@@ -1179,6 +1179,24 @@ void ModFunction::calc_unprovided_times(TimingData const* timing)
 	calc_timing_pair(timing, m_end_beat, m_end_second);
 }
 
+#define MF_NEEDS_THING(thing) \
+bool ModFunction::needs_##thing() \
```

Comment:
Minor nitpick. Should the name of the method be `needs_##_thing` to better separate the things?